api.ts

```diff
@@ -1151,6 +1172,27 @@ export interface SyncFormulaDef<
    * @hidden
    */
   updateOptions?: Pick<CommonPackFormulaDef<ParamDefsT>, 'extraOAuthScopes'>;
+
+  /**
+   * The javascript function that implements fetching permissions for a set of objects
+   * if the objects in this sync table have permissions in the external system.
+   *
+   * TODO(sam): Unhide this
+   * @hidden
+   */
+  executeGetPermissions?(
```

Comment:
Do we need a new formula type here?

Reply:
I do not think there is a real alternative after spleunking through the code in more depth